modelon/impact/client/operations/base.py

```diff
@@ -182,17 +184,24 @@ class instance if the set status is achieved.
 
         """
         start_t = time.time()
-        while True:
-            logger.info(f"{self.name} in progress! Status : {self.status.name}")
-            if self.status == status:
-                logger.info(f"{self.name} completed! Status : {self.status.name}")
-                return self.data()
-
-            current_t = time.time()
-            if timeout and current_t - start_t > timeout:
-                raise exceptions.OperationTimeOutError(
-                    f"Time exceeded the set timeout - {timeout}s! "
-                    f"Present status of operation is {self.status.name}!"
-                )
-
-            time.sleep(0.5)
+        status_tuple = status if isinstance(status, tuple) else tuple([status])
+
+        try:
+            while True:
+                logger.info(f"{self.name} in progress! Status : {self.status.name}")
+                if self.status in status_tuple:
+                    logger.info(f"{self.name} completed! Status : {self.status.name}")
+                    return self.data()
+
+                current_t = time.time()
+                if timeout and current_t - start_t > timeout:
+                    raise exceptions.OperationTimeOutError(
+                        f"Time exceeded the set timeout - {timeout}s! "
+                        f"Present status of operation is {self.status.name}!"
```

Comment:
Bake in the message into the exception. Constructor arguments should be (timeout, self.status.name, status_names).